substrate/frame/staking-async/src/slashing.rs

```diff
@@ -728,35 +394,6 @@ pub(crate) fn clear_era_metadata<T: Config>(obsolete_era: EraIndex) {
 	NominatorSlashInEra::<T>::remove_prefix(&obsolete_era, None);
 }
 
-/// Clear slashing metadata for a dead account.
-pub(crate) fn clear_stash_metadata<T: Config>(
```

Comment:
Re: https://github.com/paritytech/polkadot-sdk/pull/8316#discussion_r2073285659
@tdimitrov this is already removed.